msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/ltw/TestCase2582292.java

```diff
@@ -0,0 +1,150 @@
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.msal.automationapp.testpass.broker.ltw;
+
+import androidx.test.uiautomator.UiObjectNotFoundException;
+
+import com.microsoft.identity.client.msal.automationapp.R;
+import com.microsoft.identity.client.msal.automationapp.testpass.broker.AbstractMsalBrokerTest;
+import com.microsoft.identity.client.ui.automation.annotations.LTWTests;
+import com.microsoft.identity.client.ui.automation.annotations.SupportedBrokers;
+import com.microsoft.identity.client.ui.automation.app.MsalTestApp;
+import com.microsoft.identity.client.ui.automation.broker.BrokerLTW;
+import com.microsoft.identity.client.ui.automation.broker.BrokerMicrosoftAuthenticator;
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.MicrosoftStsPromptHandlerParameters;
+import com.microsoft.identity.labapi.utilities.client.ILabAccount;
+import com.microsoft.identity.labapi.utilities.client.LabQuery;
+import com.microsoft.identity.labapi.utilities.constants.AzureEnvironment;
+import com.microsoft.identity.labapi.utilities.constants.FederationProvider;
+import com.microsoft.identity.labapi.utilities.constants.TempUserType;
+import com.microsoft.identity.labapi.utilities.constants.UserRole;
+import com.microsoft.identity.labapi.utilities.constants.UserType;
+import com.microsoft.identity.labapi.utilities.exception.LabApiException;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.List;
+
+@LTWTests
+@SupportedBrokers(brokers = {BrokerLTW.class})
+public class TestCase2582292 extends AbstractMsalBrokerTest {
+
+    @Test
+    public void test_2582292() throws LabApiException, InterruptedException, UiObjectNotFoundException {
+        final String username = mLabAccount.getUsername();
+        final String password = mLabAccount.getPassword();
+
+        // Install new Auth app with broker SDK changes of broker selection logic
+        final BrokerMicrosoftAuthenticator brokerMicrosoftAuthenticator = new BrokerMicrosoftAuthenticator();
+        brokerMicrosoftAuthenticator.uninstall();
+        brokerMicrosoftAuthenticator.install();
+
+        // Install legacy MSAL Test app (Msal test app with no broker selection logic)
+        final MsalTestApp msalTestApp = new MsalTestApp();
+        msalTestApp.installOldApk();
+
+        // Set up shared device mode
+        // Open Authenticator app -> ... -> Settings
+        brokerMicrosoftAuthenticator.performSharedDeviceRegistration(username, password);
+
+        // Check mode in MSAL test app
+        msalTestApp.launch();
+        msalTestApp.handleFirstRun();
+        final String mode = msalTestApp.checkMode();
+        Assert.assertTrue(mode.contains("Single Account"));
+
+        // performs AcquireToken with an account from the a same tenant with the WPJed account.
```

Comment:
nit : performs acquireToken with an account from the same tenant as the WPJed account.